homeassistant/components/aosmith/sensor.py

```diff
@@ -0,0 +1,84 @@
+"""The sensor platform for the A. O. Smith integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AOSmithData
+from .const import DOMAIN, HOT_WATER_STATUS_MAP
+from .coordinator import AOSmithCoordinator
+from .entity import AOSmithEntity
+
+
+@dataclass
+class AOSmithRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[dict[str, Any]], str | int | None]
+
+
+@dataclass
```

Comment:
Do a global search for kw_only, it will save you a separate class. (You don't need the mixin anymore)

Reply:
Thanks, didn't know about this. Fixed.